server/src/main/java/org/opensearch/extensions/DiscoveryExtensionNode.java

```diff
@@ -59,6 +69,9 @@ public void writeTo(StreamOutput out) throws IOException {
         for (ExtensionDependency dependency : dependencies) {
             dependency.writeTo(out);
         }
+        for (Scope scope : scopes) {
```

Comment:
```suggestion
if (out.getVersion().onOrAfter(Version.V_3_0_0)) {
        out.writeList(scopes);
}

```

Reply:
Why add an version filter? Extensions (and scopes) are safe to backport since they are feature flagged off

Reply:
But feature flag in 2.9.0 != 3.0.0: there are new properties added and the old clusters won't be able to deal with this new data